sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,273 @@
+"""Implementations of pathways for use by actuators.
+
+Notes
+=====
+
+This module is experimental and so is named with a leading underscore to
+indicate that the API is not yet stabilized and could be subject to breaking
+changes.
+
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import S, Symbol
+from sympy.core.expr import Expr
+from sympy.physics.mechanics import Point
+from sympy.physics.vector import Vector, dynamicsymbols
+
+
+__all__ = ['LinearPathway']
+
+
+class PathwayBase(ABC):
+    """Abstract base class for all pathway classes to inherit from.
+
+    Notes
+    =====
+
+    Instances of this class cannot be directly instantiated by users. However,
+    it can be used to created custom pathway types through subclassing.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        origin: Point,
+        insertion: Point,
+    ) -> None:
+        """Initializer for ``PathwayBase``."""
+        self.origin = origin
+        self.insertion = insertion
+
+    @property
+    def origin(self) -> Point:
```

Comment:
Based on another discussion I expect these to be renamed, as `origin` and `insertion` are not from a general, but from a biomechanical perspective.